modules/infinite-scroll/infinity.php

```diff
@@ -251,7 +251,15 @@ static function get_settings() {
 	static function posts_per_page() {
 		$posts_per_page = self::get_settings()->posts_per_page ? self::get_settings()->posts_per_page : self::wp_query()->get( 'posts_per_page' );
 
-		// Take JS query into consideration here
+		// If Infinite Scroll is set to click, and if the site owner changed posts_per_page, let's use that.
+		if (
+			'click' === self::get_settings()->type
+				&& ( '10' !== get_option( 'posts_per_page' ) )
```

Comment:
Should we use `self::wp_query()->get( 'posts_per_page' )`, or `$posts_per_page` as used above? I'm not super clear on what's the difference here?

Reply:
My understanding is that  `self::get_settings()->posts_per_page` is the setting that gets declared when the theme registers support for infinite scroll, whereas `self::wp_query()->get( 'posts_per_page' )` is the number of posts that the query in the current loop is getting. The idea of this line is to get the setting from `wp-admin/options-reading.php` rather than whatever the theme registered, so that users can change it from the default if they want.

The problem with this approach is that if the theme registers a value which is not the default and the users wants the default, they won't be able to.

Reply:
We're calling `get_option` twice in short succession for the same option. Could we just put the value in a local var and use that?

Reply:
Done in 2577ee812